integration/query_comparison_compat_test.go

```diff
@@ -69,23 +69,55 @@ func TestQueryComparisonCompatImplicit(t *testing.T) {
 			filter:     bson.D{{"v.some.0", bson.A{42}}},
 			resultType: emptyResult,
 		},
+		"Int32": {
+			filter:             bson.D{{"v", int32(42)}},
+			resultPushdown:     true,
+			skipTigrisPushdown: true,
+		},
+		"Int64": {
+			filter:             bson.D{{"v", int64(42)}},
+			resultPushdown:     true,
+			skipTigrisPushdown: true,
+		},
```

Comment:
```suggestion
		},
		"Double": {
			filter:             bson.D{{"v", 42}},
			resultPushdown:     true,
			skipTigrisPushdown: true,
		},
```
This looks like interesting case, because I think it should match.

Reply:
The interesting thing here is that if int64(42) in [int64](https://github.com/FerretDB/FerretDB/blob/main/integration/shareddata/scalars.go#L97) and int32(42) in [int32](https://github.com/FerretDB/FerretDB/blob/main/integration/shareddata/scalars.go#L86) matches `float64(42)` or not. As I understand float64(42) is same as 42. We have test for double 42.13 but not for double 42.

Reply:
Actually the test I'm looking for is already there `ValueNumber` :see_no_evil: My bad